sr_gui_change_controllers/src/sr_gui_change_controllers/change_controllers.py

```diff
@@ -51,64 +57,136 @@ def __init__(self, context):
+                self.teach_mode_button_toggled_lh)
+            self._lh_teach_buttons.append(self._widget.lh_pos)
+
+            self._widget.lh_teach.setIcon(self.CONTROLLER_OFF_ICON)
+            self._widget.lh_teach.toggled.connect(
+                self.teach_mode_button_toggled_lh)
+            self._lh_teach_buttons.append(self._widget.lh_teach)
+            # hide teach mode if arm...
+            if 'la_' in self.avaliable_controller_groups:
+                self._widget.lh_teach.hide()
+
+        # TODO: add buttons when more controller modes are available for arms.
+        self._widget.ra_group.hide()
+        self._widget.la_group.hide()
+
+        self._widget.information_box.clicked.connect(self.display_information)
+        self.confirm_current_control()
+
+    def find_active_control_groups(self):
+        joint_states = rospy.wait_for_message('/joint_states', JointState)
+        avaliable_robot_names = []
+        for robot_name in self.robot_names:
+            for joint in joint_states.name:
+                if joint.startswith(robot_name):
+                    if robot_name not in avaliable_robot_names:
+                        avaliable_robot_names.append(robot_name)
+        return avaliable_robot_names
+
+    def confirm_current_control(self):
+        list_controllers = rospy.ServiceProxy(
+            'controller_manager/list_controllers', ListControllers)
+        try:
+            resp1 = list_controllers()
+        except rospy.ServiceException:
+            rospy.logerr(
+                "Couldn't get list of controllers from controller_manager/list_controllers service")
+            return
+
+        running_controllers = [c for c in resp1.controller if c.state == "running"]
+
+        current_robot_control = {
+            'rh_': None,
+            'lh_': None,
+            'ra_': None,
+            'la_': None
+        }
+        for robot_name in self.robot_names:
+            for controller in running_controllers:
+                if robot_name in controller.name:
+                    if "position_controller" in controller.name:
+                        current_robot_control[robot_name] = 1
+                        break
+                    elif "trajectory_controller" in controller.name:
+                        current_robot_control[robot_name] = 0
+                        break
+                    elif "effort_controller" in controller.name:
+                        current_robot_control[robot_name] = 2
+                        break
+
+        for robot_name, control in current_robot_control.items():
+            if control is not None:
+                self.update_current_controller_field(control, robot_name)
+
+    def update_current_controller_field(self, ctrl_type, robot_name):
+        if "rh_" == robot_name:
+            for button in range(len(self._rh_teach_buttons)):
+                if button == ctrl_type:
+                    self._rh_teach_buttons[button].setIcon(self.CONTROLLER_ON_ICON)
```

Comment:
teach_buttons variable is a bit confusing, maybe use something more rh_control_buttons or if you can think of something better (i know it was from old code).